pkg/controller/constrainttemplate/constrainttemplate_controller_test.go

```diff
@@ -510,21 +517,28 @@ func TestReconcile(t *testing.T) {
 			logger.Error(err, "create cstr")
 			t.Fatal(err)
 		}
-		// check if vapbinding resource exists now
-		vapBinding := &admissionregistrationv1beta1.ValidatingAdmissionPolicyBinding{}
-		vapBindingName := fmt.Sprintf("gatekeeper-%s", denyall+strings.ToLower(suffix))
-		if err := c.Get(ctx, types.NamespacedName{Name: vapBindingName}, vapBinding); err != nil {
-			if !apierrors.IsNotFound(err) {
-				t.Fatal(err)
+		err = retry.OnError(testutils.ConstantRetry, func(_ error) bool {
+			return true
+		}, func() error {
+			// check if vapbinding resource exists now
+			vapBinding := &admissionregistrationv1beta1.ValidatingAdmissionPolicyBinding{}
+			vapBindingName := fmt.Sprintf("gatekeeper-%s", denyall+strings.ToLower(suffix))
+			if err := c.Get(ctx, types.NamespacedName{Name: vapBindingName}, vapBinding); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+				return nil
 			}
-		} else {
-			t.Fatal("should result in error, vapbinding not found")
+			return fmt.Errorf("should result in error, vapbinding not found")
+		})
+		if err != nil {
+			t.Fatal(err)
 		}
 	})
 
-	t.Run("VapBinding should be created with v1beta1", func(t *testing.T) {
-		suffix := "VapBindingShouldBeCreatedV1Beta1"
-		logger.Info("Running test: VapBinding should be created with v1beta1")
+	t.Run("VapBinding should not be created without VAP", func(t *testing.T) {
```

Comment:
Please change all occurrances where it says "without VAP" instead it should be "without generateVap intent in CT"